taskhawk/backends/gcp.py

```diff
@@ -261,3 +262,17 @@ def requeue_dead_letter(self, num_messages: int = 10, visibility_timeout: int =
                     )
 
             logging.info("Re-queued {} messages".format(len(queue_messages)))
+
+    def health_check(self) -> None:
+        """
+        Checks that subscriber can synchronously pull messages from the queue.
+        """
+        try:
+            self.subscriber.pull(
+                subscription=self._subscription_path,
+                max_messages=1,
+                retry=None,
+                timeout=settings.GOOGLE_PUBSUB_READ_TIMEOUT_S,
+            )
+        except Exception as err:
```

Comment:
Aren't we expecting an exception to be raised when there are no messages similarly to https://github.com/cloudchacho/taskhawk-python/blob/master/taskhawk/backends/gcp.py#L184 ?

Reply:
We are mostly concerned with `ServiceUnavailable`, but the `Exception` should catch all exceptions, and we can then access the chained exception.

Reply:
Oh, do you mean we should not raise an exception for `DeadlineExceeded`?

Reply:
That is my understanding from the way the above linked code is written, but I could totally be wrong. May be worth investigating into Google's pubsub exception library

Reply:
Looking at the Cloud Pub/Sub docs, they recommend retrying https://cloud.google.com/pubsub/docs/reference/error-codes

So yes, we shouldn't treat it as an exception.